lib/symboldatabase.cpp

```diff
@@ -6398,6 +6398,16 @@ void SymbolDatabase::setValueType(Token* tok, const ValueType& valuetype, Source
                 } else if (parsedecl(vt2->containerTypeToken, &autovt, mDefaultSignedness, mSettings, mIsCpp)) {
                     setType = true;
                     templateArgType = vt2->containerTypeToken->type();
+                    if (Token::simpleMatch(autoToken->next(), "&"))
+                        autovt.reference = Reference::LValue;
+                    else if (Token::simpleMatch(autoToken->next(), "&&"))
+                        autovt.reference = Reference::RValue;
+                    if (autoToken->previous()->str() == "const") {
+                        if (autovt.pointer && autovt.reference != Reference::None)
+                            autovt.constness |= 2;
```

Comment:
should it say:

    autovt.constness |= (1 << autovt.pointer);

Reply:
Probably, but the matched string above (see TODO) only allows one `*` anyway.

Reply:
:+1: